Sources/AWSLambdaRuntimeCore/NewLambda+JSON.swift

```diff
@@ -0,0 +1,114 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftAWSLambdaRuntime open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftAWSLambdaRuntime project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+import NIOFoundationCompat
+
+import class Foundation.JSONDecoder
+import class Foundation.JSONEncoder
+
+package protocol LambdaEventDecoder {
+    func decode<Event: Decodable>(_ type: Event.Type, from buffer: ByteBuffer) throws -> Event
+}
+
+package protocol LambdaOutputEncoder {
+    func encode<Output: Encodable>(_ value: Output, into buffer: inout ByteBuffer) throws
+}
+
+extension JSONEncoder: LambdaOutputEncoder {}
+
+extension JSONDecoder: LambdaEventDecoder {}
```

Comment:
this needs to go into AWSLambdaRuntime not the Core part. We want to keep LambdaCore Foundation free.